### ISSUE TITLE
Automatically resolved path parameters are always marked as optional

### ISSUE THREAD

**Issue by jboutros** [hbakri/django-ninja-crud#471]
I see that `api_view.resolve_path_parameters` is annotated to return `Optional[Type[pydantic.BaseModel]]` since it can return None if there are no path parameters defined.

This seems to have a side effect of making all path parameters as optional (required: false) in the openapi spec. Is there any way around this other than specifying `path_parameters` explicitly on the endpoint definition?

This endpoint:
```python
    get_object = views.ReadView(
        path="/{external_id}",
        get_model=lambda request, path_parameters: ViewSet.model.objects.get(external_id=path_parameters.external_id),
        response_body=ObjectSchema,
    )
```

Results in this def:
```json
"/api/me/obj/{external_id}": {
      "get": {
        "operationId": "get_object",
        "summary": "Get Object",
        "parameters": [
          {
            "in": "path",
            "name": "external_id",
            "schema": {
              "format": "uuid",
              "title": "External Id",
              "type": "string"
            },
            "required": false
          }
        ],
        "responses": {
          "200": {
            "description": "OK",
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ObjectSchema"
                }
              }
            }
          }
        },
```

**Comment by hbakri** [comment]
Hi @jboutros, thanks for reporting this issue! While the issue isn't related to the signature of `api_view.resolve_path_parameters`, you're right that the behavior is not normal - since `v1.2`, Django Ninja has been wrapping `pk` types in `Optional` (see [`ninja/orm/fields.py`](https://github.com/vitalik/django-ninja/blob/master/ninja/orm/fields.py#L153-L155)), likely for automatic [`ModelSchema`](https://django-ninja.dev/guides/response/django-pydantic/) definition. This wasn't appropriate for path parameters, and I wasn't sure how to address it. I've now found a solution (cf. #475) to make all automatically inferred path parameters mandatory (which aligns with RESTful API conventions where optional parameters should be query parameters):

```python
schema_fields: Dict[str, Any] = {}
for field_name in path_parameters_names:
    model_field = self.model._meta.get_field(field_name)
    schema_field = ninja.orm.fields.get_schema_field(field=model_field)[0]
    schema_fields[field_name] = (
        typing.get_args(schema_field)[0]
        if typing.get_origin(schema_field) is Union
        else schema_field,
        ...,
    )
```

This ensures that automatically resolved path parameters aren't wrapped in `Optional` anymore, correctly marking them as required in the OpenAPI spec.

PS: Unrelated to this issue but worth noting: your custom `get_model` implementation is actually the default behavior. You can simplify your code by removing this parameter, as `ReadView` already implements this logic:

```python
def _default_get_model(
    self, request: HttpRequest, path_parameters: Optional[BaseModel]
) -> Model:
    return self.model.objects.get(
        **(path_parameters.model_dump() if path_parameters else {})
    )
```

I'll include this fix in the next release `v0.6.2`, which should be available tonight or tomorrow. Thanks for helping improve Django Ninja CRUD! 🌞

**Comment by hbakri** [comment]
Hi @jboutros, great news! I've just released `v0.6.2` with the fix. All path parameters should now be correctly marked as required in the OpenAPI spec. Check out the release notes here: https://github.com/hbakri/django-ninja-crud/releases/tag/v0.6.2

Thanks again for your help in improving Django Ninja CRUD! Please let me know if you encounter any issues with the new version or anything else.

**Comment by jboutros** [comment]
thank you @hbakri I look forward to trying this out. And thank you for all of your hard work creating and maintaining such a useful library!
